spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -197,4 +198,87 @@ static Schema getOutputSchema(Schema inputSchema, String predictionField) {
     return Schema.recordOf(inputSchema.getRecordName() + ".predicted", fields);
   }
+
+  /**
+   * Validate config parameters for feature generator classes.
+   *
+   * @param inputSchema         input Schema
+   * @param outputColumnMapping List of the input fields to map to the transformed output fields.
+   */
+  static void validateFeatureGeneratorConfig(Schema inputSchema, String outputColumnMapping) {
+    Map<String, String> map;
+    try {
+      map = Splitter.on(',').trimResults().withKeyValueSeparator(":").split(outputColumnMapping);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(
+        String.format("Invalid output feature mapping. %s. Please make sure it is in the format " +
+                        "'input-column':'transformed-output-column'.", e.getMessage()), e);
+    }
+    for (Map.Entry<String, String> entry : map.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      if (inputSchema.getField(key) == null) {
+        throw new IllegalArgumentException(String.format("Input field %s does not exist in the input schema %s.",
+                                                         key, inputSchema.toString()));
+      }
+      Schema schema = inputSchema.getField(key).getSchema();
+      Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+      Schema.Type componentSchemaType = null;
+      if (type.equals(Schema.Type.ARRAY)) {
+        Schema componentSchema = schema.getComponentSchema();
+        componentSchemaType = componentSchema.isNullable() ? componentSchema.getNonNullable().getType() :
+          componentSchema.getType();
+      }
+      if (!(type.equals(Schema.Type.STRING) || (type.equals(Schema.Type.ARRAY) &&
+        componentSchemaType.equals(Schema.Type.STRING)))) {
+        throw new IllegalArgumentException(String.format("Field to be transformed should be of type String or " +
+                                                           "Nullable String or Array of type String or Nullable " +
+                                                           "String . But was %s for field %s.", type, key));
+      }
+      if (value == null) {
+        throw new IllegalArgumentException(String.format("Output column name not specified for column : %s. " +
+                                                           "Please make sure it is in the format 'input-column':" +
+                                                           "'transformed-output-column'.", entry.getKey()));
+      } else if (inputSchema.getField(value) != null) {
+        throw new IllegalArgumentException(
+          String.format("Output column name %s for column %s is already present in the input schema. Please " +
+                          "provide a different output field name.", value, key));
+      }
+    }
+  }
+
+  /**
+   * Get the valid of input field which is to be used for text based feature generation.
```

Comment:
Gets the input field for feature generation. If the field is an array, returns the field. Otherwise, splits the field based on the given pattern. Assumes the input field is an array of strings or a string.